dev/src/reference.ts

```diff
@@ -1725,22 +1775,33 @@ class QueryUtil<
             docs.reverse();
           }
 
-          resolve(
-            query._createSnapshot(
-              readTime,
-              docs.length,
-              () => docs,
-              () => {
-                const changes: Array<
-                  DocumentChange<AppModelType, DbModelType>
-                > = [];
-                for (let i = 0; i < docs.length; ++i) {
-                  changes.push(new DocumentChange('added', docs[i], -1, i));
-                }
-                return changes;
+          // TODO(ehsannas): Typings of QuerySnapshot suggest that readTime is always available
```

Comment:
I talked to @ehsannas and read through some documentation.

```
// If the query returns no results, a response with `read_time` and no
// `document` will be sent, and this represents the time at which the query
// was run.
```
 
 Note, `readTime` represents the time at which query was run. So, when calling `explain(options)` with `options.analyze !== true`, indicating NO execution is to be done, ONLY a plan is to be returned, then `readTime` will be absent, but we also omit the snapshot from result.